File: problems/Min Cost Climbing Stairs/minCostClimbingStairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int downOne = 0;
        int downTwo = 0;

        for (int idx = 2; 1 + cost.length > idx; ++idx) {
            int tmp = downOne;

            downOne = Math.min(downOne + cost[idx - 1], downTwo + cost[idx - 2]);
            downTwo = tmp;
        }

        return downOne;
    }
}

File: problems/Min Cost Climbing Stairs/minCostClimbingStairs.cs
public class Solution {
    public int MinCostClimbingStairs(int[] cost) {
        var dp = new int[1 + cost.Length];

        for (var i = 2; cost.Length >= i; ++i) {
            dp[i] = Math.Min(cost[i - 2] + dp[i - 2], cost[i - 1] + dp[i - 1]);
        }

        return dp[cost.Length];
    }
}
